src/lock_hash.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+"""Charm code for MongoDB service."""
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import os
+from logging import getLogger
+
+import ops
+
+from config import Config
+from machine_helpers import ROOT_USER_GID
+
+logger = getLogger(__name__)
+
+HASH_KEY = "lockhash"
+LOCK_PATH = Config.MONGODB_DATA_DIR / f".{HASH_KEY}"
+UNDEFINED = "UNDEFINED"
+
+
+class LockHashHandler:
+    """Descriptor class for the lock hash stored in the file."""
```

Comment:
I think it would be VERY useful in the docstring for this had context on why (in the charming sense) this is necessary